string[] locations
string[] tasks
---
string result
